### PR TITLE
better initialization patterns for seminst

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: compiler/seminst.nim [hunk 1]
```diff
@@ -359,8 +351,7 @@ proc generateInstance(c: PContext, fn: PSym, pt: TIdTable,
     globalError(c.config, info, "generic instantiation too nested")
   n[namePos] = newSymNode(result)
   pushInfoContext(c.config, info, fn.detailedInfo)
-  var entry = TInstantiation.new
-  entry.sym = result
+  var entry = TInstantiation(sym: result)
```
Comment (ringabout) [inline compiler/seminst.nim]:
```suggestion
  var entry = TInstantiation.new
  entry.sym = result
```

**Comment by blackmius** [comment]
wouldn't it be better to find all such initializations using grep and change them with one commit?

``` sh
with default initialization
grep -r 'var \w*: \w* = default' *

without init
grep -r 'var \w*: [A-Z]\w*$' *

grep -r 'new(' *

check again after limitating var x: X = default(X)
grep -r 'default(' *
```

**Comment by ringabout** [comment]
Well, I did use grep tools to find the usage. But I might or might not combine them into one PR. It's not a big deal after all. Besides, it's easier to review.

File: compiler/seminst.nim [hunk 2]
```diff
@@ -29,10 +29,7 @@ proc addObjFieldsToLocalScope(c: PContext; n: PNode) =
   else: discard
 
 proc pushProcCon*(c: PContext; owner: PSym) =
-  var x: PProcCon
-  new(x)
-  x.owner = owner
-  x.next = c.p
+  let x: PProcCon = PProcCon(owner: owner, next: c.p)
   c.p = x
```
Comment (ringabout) [inline compiler/seminst.nim]:
```suggestion
  c.p = PProcCon(owner: owner, next: c.p)
```

**Comment by blackmius** [comment]
> 

We just don't know how many more such PRs there will be and when it's worth saying "there is still a use here that needs to be corrected"

**Comment by ringabout** [comment]
> needs to be corrected"

There is a problem with object initialization, which is a bug of strict assignment analysis. Other refactorings are more like a style change. I'm just relieving the arduous work of the future me. It's really pesky to have all things done at once.
 
```nim
  var x: TProcCon
  x.owner = owner
  x.next = c.p
  c.p = x
```
I will look into it when I finish my main work of the start is https://github.com/nim-lang/Nim/pull/22446

**Pull request merged**